### ISSUE TITLE
[Due for payment 2026-01-30] Screen Readers: Settings Security: The fields of code are announced without a programmatic name

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#76912]
## Action Performed:
Prerequisites:
- the user is signed in

1. Using Chrome + JAWS, open the URL – https://new.expensify.com/
2. Navigate using the Tab key to the Account button and press Enter to activate it.
3. Navigate using the Tab key to the Security button and press Enter.
4. Navigate to the Two-factor authentication section and press Enter.
5. Enter the verification code that is sent to your registered Email.
6. In the Two-factor authentication modal, fill in the steps and click the Next button.
7. In step 2, press the Tab key to focus the 6 fields for the code.
8. Note how each field is announced.

## Expected Result:
Each code input field should have a programmatic name (label or aria-label) that clearly indicates its purpose, such as:
Each input field representing a digit in the six-digit authentication code should be announced with a programmatic name, e.g.:
- 'enter digit 1 of 6'
- 'enter digit 2 of 6'
- 'enter digit 3 of 6'
- 'enter digit 4 of 6'
- 'enter digit 5 of 6'
- 'enter digit 6 of 6'

## Actual Result:
The code input fields are announced with no identifying label or context. The screen reader does not announce what the field is for, causing confusion for users.

HTML:
<input autocomplete="one-time-code" autocorrect="on" placeholder="" maxlength="6" autocapitalize="sentences" dir="auto" inputmode="numeric" rows="1" spellcheck="true" virtualkeyboardpolicy="auto" role="presentation" class="css-11aywtz r-6taxm2 r-13awgt0 r-13qz1uu r-1yadl64 r-vzf2wf r-a023e6 r-13uqrnb r-16dba41 r-rjixqe r-gy4na3 r-8b1cjr r-van48c r-fpub7 r-p30vwv r-1j63xyz r-1mdbw0j r-wk8lta r-105ug2t fs-mask" data-testid="twoFactorAuthCodeInput" data-component="AnimatedComponent" data-element="Component" data-source-file="createAnimatedComponent.js" type="text" value="" style="--placeholderTextColor: #8B9C8F; outline-width: 0px; outline-style: none; box-shadow: none;">

## Area issue was found in:
Settings-Security

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
If each digit is required to be entered into a separate field, use aria-label to announce each field, e.g.:
- 'enter digit 1 of 6'
- 'enter digit 2 of 6'
- 'enter digit 3 of 6'
- 'enter digit 4 of 6'
- 'enter digit 5 of 6'
- 'enter digit 6 of 6'

## Environment(s)
Windows Windows 11 - Chrome

## Screenshots/Videos
<img width="1919" height="922" alt="Image" src="https://github.com/user-attachments/assets/b3985b28-3ec6-4379-92fa-cb2fef523dd5" />

https://github.com/user-attachments/assets/df5a9d58-67c4-4dc0-8a99-a7fde1826abd

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6999182
**Issue reported by:** NA
**Slack conversation:** NA

**Comment by MelvinBot** [comment]
Bug reproducible - https://platform.applause.com/services/links/v1/external/00378014245b8ee54a65c901931613658fcc8f856f2fbc5cb59f3a13a2694216
